### PR TITLE
docs: add Redis Streams Source doc

### PR DESCRIPTION
**Pull request by juliev0**
Add a doc for how to use Redis Streams Source.

### CODE REVIEW

File: docs/user-guide/sources/redis-source.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          stream: test-stream
+          consumerGroup: my-group
+          readFromBeginning: true # Should we start from beginning of Stream or latest?
+
+```
+
+Please see [API](https://github.com/numaproj/numaflow/blob/main/docs/APIs.md#redisstreamssource) for details on how to optionally do the following:
+- Define TLS
+- Define username/password
+- Connect to Redis Sentinel
```
Comment (juliev0) [inline docs/user-guide/sources/redis-source.md:23]: Let me know if you prefer that I show some subset of this as an example.

File: test/redis-source-e2e/testdata/redis-source-pipeline-from-beginning.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ spec:
         min: 2
       source:
         redisStreams:
-          url: "redis:6379"
+          url: redis:6379
           stream: test-stream-a
```
Comment (juliev0) [inline test/redis-source-e2e/testdata/redis-source-pipeline-from-beginning.yaml:17]: thought I'd remove the unnecessary quotes

**Comment by whynowy** [comment]
Please add it to `mkdocs.yml`

**Comment by juliev0** [comment]
> Please add it to `mkdocs.yml`

ahh, forgot about that

**Review by whynowy — APPROVED** [review]

**Pull request merged**